internal/receiver/discoveryreceiver/evaluator.go

```diff
@@ -83,7 +84,14 @@ func (e *evaluator) evaluateMatch(match Match, pattern string, status discovery.
 		env := e.exprEnv(pattern)
 		env["ExprEnv"] = env
 		// TODO: cache compiled programs for performance benefit
-		if program, err = expr.Compile(match.Expr, expr.Env(env)); err != nil {
+		program, err = expr.Compile(
```

Comment:
We should update the docs [here](https://github.com/signalfx/splunk-otel-collector/tree/main/internal/receiver/discoveryreceiver#expr) with a note about `type` being moved to `typeOf` so users have the option to use the function in their expr.

Reply:
[added a unit test](https://github.com/signalfx/splunk-otel-collector/pull/3559/commits/9eca2da8ba09f4ddebba2d258e9bb643e82cc251) to be certain typeOf is usable

Reply:
Added a note in the README.